clearwater-diags-monitor/usr/share/clearwater/bin/clearwater_diags_monitor

```diff
@@ -738,10 +738,37 @@ do
   # -z  Zip the archive.
   # -f  Name of the archive.
   (cd $DUMPS_DIR; tar -pcz -f $CURRENT_DUMP_ARCHIVE $CURRENT_DUMP)
-  log "Diagnostic archive $CURRENT_DUMP_ARCHIVE created"
-  rm -rf "$CURRENT_DUMP_DIR"
 
-  mv $CURRENT_DUMP_ARCHIVE $FINAL_DUMP_ARCHIVE
+  # Check whether the tar command succeeded
+  ret_code=$?
+  if [ $ret_code == 0 ]; then
+    log "Diagnostic archive $CURRENT_DUMP_ARCHIVE created"
+    rm -rf "$CURRENT_DUMP_DIR"
+    mv $CURRENT_DUMP_ARCHIVE $FINAL_DUMP_ARCHIVE
+  else
+    # We failed to create the archive. The most common cause of this is lack of
+    # disk space.
```

Comment:
Probably worth deleting the CURRENT_DUMP file if it exists.

Reply:
Done